### PR TITLE
F/ddos protection

### PR DESCRIPTION
**Pull request by sa-progress**
### Description

Please describe what this change achieves. Ensure you have read the [Contributing to InSpec](https://github.com/inspec/inspec-azure/CONTRIBUTING.MD) document before submitting.

### Issues Resolved

List any existing issues this PR resolves, or any Discourse or StackOverflow discussion that's relevant

### Check List

- [x] New functionality includes tests
- [x] All tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: docs/resources/azure_ddos_protection_resource.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ end
+ ```
+## Parameters
+
+| Name                           | Description                                                                      |
+ |--------------------------------|----------------------------------------------------------------------------------|
+| resource_group                 | Azure resource group that the targeted resource resides in. `MyResourceGroup`    |
+| name                           | Name of the Azure resource to test. `Myddos_protection_plan`                          |
+| type                           | type of ddos_protection_plan                                                          |
+| provisioning_state             | State of ddos_protection_plan creation                                                |
```
Comment (sathish-progress) [inline docs/resources/azure_ddos_protection_resource.md:43]: why are these in parameters?

File: docs/resources/azure_ddos_protection_resources.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+ end
+ ```
+
+## Properties
+
+|Property       | Description                                                                          | Filter Criteria<superscript>*</superscript> |
+ |---------------|--------------------------------------------------------------------------------------|-----------------|
+| name           | A list of the unique resource names.                                                | `name`            |
+| ids            | A list of ddos_protection_plan ids .                                                       | `id`              |
+| virtual_networks           | The list of virtual networks associated with the DDoS protection plan resource. This list is read-only.| `virtual_network`             |
```
Comment (sathish-progress) [inline docs/resources/azure_ddos_protection_resources.md:45]: as per [this](https://github.com/inspec/inspec-azure/pull/415/files#diff-e5245bbcb7b027855cfdee2d684821191de17744f67f370beb77b710ee5e1bd0R36) field should be `virtual_networks`

File: libraries/azure_ddos_protection_resource.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  def provisioning_state
+    properties.provisioningState if exists?
+  end
+
+  def virtual_networks
+    properties.provisioningState.virtualNetworks if exists?
+  end
+
+  def resource_guid
+    properties.provisioningState.resourceGuid if exists?
```
Comment (sathish-progress) [inline libraries/azure_ddos_protection_resource.rb:67]: as per [this](https://docs.microsoft.com/en-us/rest/api/virtualnetwork/ddos-protection-plans/get#ddosprotectionplan) it should be attrs of `properties` and not `provisioningState`

File: libraries/backend/azure_environment.rb [hunk 4]
```diff
@@ -98,8 +98,8 @@ def initialize(options)
                                                            active_directory_graph_api_version: '2013-04-05',
                                                            storage_endpoint_suffix: '.core.windows.net',
                                                            key_vault_dns_suffix: '.vault.azure.net',
-                                                           datalake_store_filesystem_endpoint_suffix: 'azuredatalakestore.net',
-                                                           datalake_analytics_catalog_and_job_endpoint_suffix: 'azuredatalakeanalytics.net',
+                                                           datalake_store_filesystem_endpoint_suffix: '.azuredatalakestore.net',
```
Comment (sathish-progress) [inline libraries/backend/azure_environment.rb:101]: this is not relevant here

File: libraries/backend/helpers.rb [hunk 5]
```diff
@@ -92,6 +92,7 @@ class AzureEnvironments
       active_directory_endpoint_url: MicrosoftRestAzure::AzureEnvironments::AzureCloud.active_directory_endpoint_url,
       storage_endpoint_suffix: MicrosoftRestAzure::AzureEnvironments::AzureCloud.storage_endpoint_suffix,
       key_vault_dns_suffix: MicrosoftRestAzure::AzureEnvironments::AzureCloud.key_vault_dns_suffix,
+      datalake_store_filesystem_endpoint_suffix: MicrosoftRestAzure::AzureEnvironments::AzureCloud.datalake_store_filesystem_endpoint_suffix,
```
Comment (sathish-progress) [inline libraries/backend/helpers.rb:95]: same again this is not relevant here

File: libraries/backend/azure_environment.rb (hunk 4 above)
Comment (sa-progress) [reply libraries/backend/azure_environment.rb:101]: reverted from master

File: libraries/backend/helpers.rb (hunk 5 above)
Comment (sa-progress) [reply libraries/backend/helpers.rb:95]: reverted from master

File: docs/resources/azure_ddos_protection_resource.md (hunk 1 above)
Comment (sathish-progress) [reply docs/resources/azure_ddos_protection_resource.md:43]: type & provisioning_state can't be params

**Review by sathish-progress — APPROVED** [review]
LGTM

**Review by IanMadd — APPROVED** [review]

**Pull request merged**